sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/SharedThriftServer.scala

```diff
@@ -138,10 +138,15 @@ trait SharedThriftServer extends SharedSparkSession {
     sqlContext.setConf(ConfVars.HIVE_START_CLEANUP_SCRATCHDIR.varname, "true")
 
     try {
-      hiveServer2 = HiveThriftServer2.startWithContext(sqlContext)
+      hiveServer2 = HiveThriftServer2.startWithContext(sqlContext, exitOnError = false)
```

Comment:
shall we add some comments to explain why we don't want to exit on error here?

Reply:
OK. Please check whether the comments added are informative or not.